### PR TITLE
Updated Worker with support for custom auth objects

### PR DESCRIPTION
**Pull request by Obdam**
### Proposed changes

* Added the `auth` headers to the instantiation of the OpenCTIApiClient;
* Created support for custom headers in the worker configuration/environment variables.

### Related issues

* This change is related to the addition of an authentication class to the client-python repository, which can be found here: https://github.com/OpenCTI-Platform/client-python/pull/483

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments
N/A.

### CODE REVIEW

File: opencti-worker/src/worker.py [hunk 1]
```diff
@@ -439,6 +440,16 @@ def __post_init__(self) -> None:
         self.log_level = get_config_variable(
             "WORKER_LOG_LEVEL", ["worker", "log_level"], config
         )
+        # Custom authentication
+        self.custom_headers = get_config_variable(
+            "CUSTOM_HEADERS", ["opencti", "custom_headers"], config
```
Comment (aHenryJard) [inline opencti-worker/src/worker.py:445]: Could you give a use case example of custom header ? I don't see "CUSTOM_HEADERS" usage in OpenCTI, so I have difficulties to see how to tests this part.

Comment (Obdam) [reply opencti-worker/src/worker.py:445]: Hi @aHenryJard,

Thanks for your reply. A potential use case is as follows:

- You're hosting the OpenCTI Platform and its Workers on a 'serverless' environment at a cloud vendor, e.g. Google Cloud's Cloud Run or Azure's Container Apps.
- Serverless environments may contain multiple authentication methods/settings, for instance an Identity-Aware Proxy.
- For instance, GCP Cloud Run expects an extra header beside the traditional `Authorization: Bearer ID_TOKEN`, namely the `X-Serverless-Authorization: Bearer ID_TOKEN`. This is further described [here](https://cloud.google.com/run/docs/authenticating/service-to-service#acquire-token).
- Therefore, you may want to influence the way how the Worker sends HTTP request to the Platform.

**Comment by Obdam** [comment]
@labo-flg What's the reason for closure?

**Comment by richard-julien** [comment]
Hi @Obdam , will be done with https://github.com/OpenCTI-Platform/client-python/issues/844.
Changing the concept of a global auth override to specify custom headers on connector and worker.

**Pull request closed without merging**